lntest/node.go

```diff
@@ -1840,6 +1859,10 @@ func (hn *HarnessNode) receiveTopologyClientStream(
 			// End of subscription stream. Do nothing and quit.
 			return nil
 
+		case errors.As(err, &context.Canceled):
```

Comment:
Fixed!

Reply:
I just realized I did try with `errors.Is` before and failed. So what's wrong if the `err` here is parsed into the variable? If the type assertion fails, it won't affect the original `err`. If it succeeds, then we exit here as it's a context canceled error?

Reply:
I think this changes the value of the global variable `context.Canceled` if it succeeds, since `As()` actually manipulates the target. It checks whether the error can be cast into the target (which in your case is just an `*error`, which succeeds) and then sets the value. So this will actually succeed with any error.
What exactly happens with `errors.Is()`? Does it just not go into the case? Then it might be a gRPC wrapped error that needs to be string matched as well...

Reply:
I got it wrong...for some reason I thought we use the target to manipulate the `err`🙈 Now switched to string match.